pkg/webhook/admission/webhook.go

```diff
@@ -114,6 +118,13 @@ func (f HandlerFunc) Handle(ctx context.Context, req Request) Response {
 	return f(ctx, req)
 }
 
+// WithHandlerFactory creates a new Webhook for a handler factory.
+func WithHandlerFactory(scheme *runtime.Scheme, obj runtime.Object, factory HandlerFactory) *Webhook {
```

Comment:
Why is this called Factory?

Reply:
Because it knows how to create a handler. It allows you to receive a decoder from the WebhookBuilder.

Another option could be just to accept a Handler (renaming the method to `WithMutationHandler`) and pass the decoder through the context.

Reply:
I decided to remove the factory and leave it to the implementation to decide how to build the object.